rust/headless-client/src/standalone.rs

```diff
@@ -192,7 +192,10 @@ pub fn run_only_headless_client() -> Result<()> {
                     }) => return Err(anyhow!(error_msg).context("Firezone disconnected")),
                     InternalServerMsg::Ipc(IpcServerMsg::Ok)
                     | InternalServerMsg::Ipc(IpcServerMsg::OnTunnelReady)
-                    | InternalServerMsg::Ipc(IpcServerMsg::OnUpdateResources(_)) => {}
+                    | InternalServerMsg::Ipc(IpcServerMsg::OnUpdateResources(_)) => {
+                        // On every resources update, flush DNS to mitigate <https://github.com/firezone/firezone/issues/5052>
+                        dns_controller.flush()?;
+                    }
```

Comment:
The callback handler code here is duplicated between the IPC service and the headless Client. Since they do almost the same thing, we can probably refactor it someday so that they share the same main loop, and the only difference is that one listens for IPC events and one listens for in-process commands